assets/js/invidious_iframe_api.js

```diff
@@ -0,0 +1,386 @@
+        }
+        else{
+            this.error_code = 2;
+            this.event_executor('error');
+        }
+        var search_params = new URLSearchParams('');
+        search_params.append('widgetid',invidious_embed.widgetid);
+        this.widgetid = invidious_embed.widgetid;
+        invidious_embed.widgetid++;
+        search_params.append('origin',location.origin);
+        search_params.append('enablejsapi','1');
+        if(typeof options.playerVars==='object'){
+            this.option_playerVars = options.playerVars;
+            for(var x in options.playerVars){
+                if(typeof x==='string'&&typeof options.playerVars[x]==='string'){
+                    search_params.append(x,options.playerVars[x]);
+                }
+            }
+            if(options.playerVars.autoplay===undefined){
+                search_params.append('autoplay','0');
+            }
+        }
+        iframe_src += "?" + search_params.toString();
+        if(options.events!==undefined&&typeof options.events==='object'){
+            for(let x in options.events){
+                this.addEventListener(x,options.events[x]);
+            }
+        }
+        this.player_iframe = document.createElement("iframe");
+        this.loaded = false;
+        this.addEventListener('loadedmetadata',()=>{this.event_executor('ready');this.loaded=true});
+        this.addEventListener('loadedmetadata',()=>{this.setVolume(this.volume)});
+        this.player_iframe.src = iframe_src;
+        if(options.width!==undefined&&typeof options.width==='number'){
+            this.player_iframe.width = options.width;
+        }
+        else{
+            if(document.body.clientWidth < 640){
+                this.player_iframe.width = document.body.clientWidth;
+            }
+            else{
+                this.player_iframe.width = 640;
+            }
+        }
+        if(options.width!==undefined&&typeof options.width==='number'){
+            this.player_iframe.width = options.width;
+        }
+        else{
+            this.player_iframe.height = this.player_iframe.width * (9/16);
+        }
+        this.player_iframe.style.border = "none";
+        replace_elemnt.replaceWith(this.player_iframe);
+        this.eventdata = {};
+        return this;
+    }
+    postMessage(data){
+        var additionalInfo = {'origin':location.origin,'widgetid':String(this.widgetid),'target':'invidious_control'};
+        data = Object.assign(additionalInfo,data);
+        this.player_iframe.contentWindow.postMessage(data,this.target_origin);
+    }
+    event_executor(eventname){
+        var execute_functions = this.eventobject[eventname];
+        var return_data = {data:undefined,target:this};
+        if(eventname==='statechange'){
+            return_data.data = this.getPlayerState();
+        }
+        for(var x=0;x<execute_functions.length;x++){
+            try{
+                execute_functions[x](return_data);
+            }
+            catch{}
+        }
```

Comment:
```suggestion
        execute_functions.forEach(func => func(return_data)};
```
Empty `try-catch` is not good.

Reply:
Your commit suggestion code has a problem.
When execute function try to execute throw error function (such as throw "error"), it will prevent other function especially behind element of execute_functions array.